### ISSUE TITLE
Match field should be a hash

### ISSUE THREAD

**Issue by joshuaspence** [logstash-plugins/logstash-filter-date#71]
It seems really strange to me that the `match` field accepts an array rather than a hash. It is also inconsistent with some other filters, such as the `grok` filter, which accepts a hash for the `match` field. Basically, this syntax seems very strange to me:

```
date {
  match => ["timestamp" , "YYYY/MM/dd HH:mm:ss"]
  remove_field => ["timestamp"]
}
```

It feels much more intuitive to configure the filter as follows:

```
date {
  match => {
    "timestamp" => "YYYY/MM/dd HH:mm:ss"
  }
  remove_field => ["timestamp"]
}
```

**Comment by jordansissel** [comment]
I agree that it is strange. The reason for the syntax is historical and continues to exist to avoid breaking backwards compatibility.

If we can come up with a way to help users upgrade to the new syntax, I'm open to this change.
